File: practica2_844759_845647/java/Read.java
//------------------------------------------------------------------------------
// File:   read.java
// Authors: Jorge Leris - 845647 / Angel Villanueva - 844759
// Date:   febrero 2023
// Coms:   Práctica 2 de TecPro
//------------------------------------------------------------------------------
import java.util.Scanner;
import java.util.Stack;

public class Read extends Instruccion
{
    public void ejecutar(Stack<Integer> Pila, int contador){

        Scanner scanner = new Scanner(System.in);
        int dato;
        
        System.out.print("? ");
        dato = scanner.nextInt();
        pila.push(dato);
        contador = contador + 1;
    }

    public void listar(){
        System.out.println("read");
    }
}

File: practica2_844759_845647/c++/read.cc
//------------------------------------------------------------------------------
// File:   read.cc
// Authors: Jorge Leris - 845647 / Angel Villanueva - 844759
// Date:   febrero 2023
// Coms:   Fichero "read" de la práctica 2
//------------------------------------------------------------------------------
#include "read.h"

void Read::ejecutar(stack<int> &pila, int &contador) const
{
    int dato;
    cout << "? ";
    cin >> dato;
    pila.push(dato);
    contador++;
}

void Read::listar() const
{
    cout << "read" << endl;
}